### ISSUE TITLE
Quitting Game Freezes

### ISSUE THREAD

**Issue by awkwaRdcaRRot** [McJty/RFToolsDimensions#62]
Every time I go to save and quit my game it freezes and I have to force close it. I removed Dimensions and it is able to shut down just fine. This is the only kind of report that I have to offer.

http://pastebin.com/J9tXcvB9

**Comment by AstralSorceress** [comment]
I do not believe that this is an issue with RFTools Dimensions (at least it is no more likely to be causing the issue than any other mod). Upon examining the [crash log](http://pastebin.com/qxfD7Y8b), you can see that FML finishes sending the ServerStopped event to all mods before Minecraft hangs. The reason that the last line in @awkwaRdcaRRot's log file is from RFTools Dimensions is that RFTools Dimensions was the last mod to print anything at log level INFO or higher (i.e. more serious, not sure if that is referred to as higher or lower), and @awkwaRdcaRRot's log file does not contain lower (higher?) log levels. With the DEBUG and TRACE levels, it is revealed that FML finished sending the ServerStopped event to all mods.

Also, the problem only occurs intermittently for me (with RFTools Dimensions, I have not tried removing it), which could explain why it appeared to subside upon the removal of RFTools Dimensions. @awkwaRdcaRRot, have you experienced the problem since removing RFTools Dimensions? Could you try adding it back in and testing if the problem occurs every time with it?

**Comment by awkwaRdcaRRot** [comment]
I haven't experienced the issue since removing it, but I also haven't experienced it on my other pack which still has it. Not sure what I changed on either since this. I had actually forgotten this was even opened. I should mention that I haven't played the pack with dimensions in it very much over the past month so I can't say for sure that the issue has subsided.

**Comment by usafphoenix** [comment]
i get this all the time in stoneblock modpack on twitch/curse platform. but as astralsorcerer mentioned, the only indication of it being tied to RFTools Dimensions is that it's the last mod that prints to the log. This isn't enough information that just the last info is "Cleaning up RFTools dimensions" it happens about 80% of the time. This could be a memory-allocation issue either tied to java itself, or the launcher, or could be specific to a mod (or a combination of both---ie: the mod only only causes quit-freezing when there isn't enough memory available). but i've also had it where i can quit and it loads the main screen fine. then i backup-overwrite a save file....load it up....play a bit, and then quit and repeat.. and it didn't crash any of the dozen or so times i quit and reloaded the same save. 
...this could be save specific, or event_triggered...where if you try and quit a game at a certain point in a mod's processing it freaks out. idk

**Comment by josephcsible** [comment]
@usafphoenix please post your whole log. This often isn't caused by the mod that happened to write the last log entry.

**Comment by usafphoenix** [comment]
size exceeds pastebin limit. only last half of log being submitted... 
https://pastebin.com/FYhugek2
